src/V3Param.cpp

```diff
@@ -1302,6 +1302,11 @@ class ParamVisitor final : public VNVisitor {
         return false;
     }
     void visit(AstVarXRef* nodep) override {
+        if (nodep->containsGenBlock()) {
+            // Needs relink, as may remove pointed-to var
+            nodep->varp(nullptr);
+            return;
+        }
```

Comment:
Where moved the behavior is slightly different since there are "return" below that used to cause it to skip this, but now will not.  However I think the logic is sound as you wrote, so let's go with it.